### PR TITLE
Add fsPromise.readdir with minimal implementation for file types

### PR DESCRIPTION
**Pull request by karthik2804**
Implemented a minimal read directory based on `nodejs`. I also added the option to request for filetypes. It does not have all the properties provided by the node but supports the major ones. 

While implementing the read directory I was wondering if we should implement `glob` as well as that seems to be straightforward to implement and might be useful.

Signed-off-by: karthik Ganeshram <karthik.ganeshram@fermyon.com>

### CODE REVIEW

File: crates/spin-js-engine/src/js_sdk/modules/fsPromises.ts [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
 /** @internal */
 const fsPromises = {
-    readFile: (filename: string) =>  {
+    readFile: (filename: string) => {
         return Promise.resolve(_fsPromises.readFile(filename))
+    },
+    readdir: (dirname: string, options?: ReadDirOptions) => {
+        if (options && options.withFileTypes == true) {
+            let res = _fsPromises.readDir(dirname, options)
+            // let ret: Array<ReadDirWithFileTypes> = []
```
Comment (dicej) [inline crates/spin-js-engine/src/js_sdk/modules/fsPromises.ts]: Can we just remove this commented-out code?

File: crates/spin-js-engine/src/lib.rs [hunk 2]
```diff
@@ -143,19 +145,97 @@ fn spin_send_http_request(context: &Context, _this: &Value, args: &[Value]) -> R
...
+                    context.wrap_callback({
+                        let e = e.clone();
+                        move |context, _, _| context.value_from_bool(e.file_type()?.is_dir())
+                    })?,
+                )?;
+
+                entry.set_property(
+                    "isSymbolicLink",
+                    context.wrap_callback({
+                        let e = e.clone();
```
Comment (dicej) [inline crates/spin-js-engine/src/lib.rs]: Can you run `cargo clippy` on this code?  I think `clippy` will tell you this clone is unnecessary.

Comment (karthik2804) [reply crates/spin-js-engine/src/lib.rs]: Will check

Comment (karthik2804) [reply crates/spin-js-engine/src/lib.rs]: `clippy` does not say anything about those clones. It did find a couple of other unnecessary stuff that I have fixed.

File: crates/spin-js-engine/src/js_sdk/modules/fsPromises.ts (hunk 1 above)
Comment (karthik2804) [reply crates/spin-js-engine/src/js_sdk/modules/fsPromises.ts]: done

**Review by dicej — APPROVED** [review]

**Pull request merged**